mmdet/datasets/coco_panoptic.py

```diff
@@ -0,0 +1,572 @@
+        'pavement-merged', 'mountain-merged', 'grass-merged', 'dirt-merged',
+        'paper-merged', 'food-other-merged', 'building-other-merged',
+        'rock-merged', 'wall-other-merged', 'rug-merged'
+    ]
+
+    def __init__(self, formator='{:012}.png', write_png=False, **kwargs):
+        self.formator = formator
+        self.write_png = write_png
+        super(CocoPanoptic, self).__init__(**kwargs)
+
+    def load_annotations(self, ann_file):
+        """Load annotation from COCO Panoptic style annotation file.
+
+        Args:
+            ann_file (str): Path of annotation file.
+
+        Returns:
+            list[dict]: Annotation info from COCO api.
+        """
+        self.coco = COCO(ann_file)
+        self.cat_ids = self.coco.get_cat_ids()
+        self.cat2label = {cat_id: i for i, cat_id in enumerate(self.cat_ids)}
+        self.categories = self.coco.cats
+        self.img_ids = sorted(self.coco.get_img_ids())
+        data_infos = []
+        for i in self.img_ids:
+            info = self.coco.load_imgs([i])[0]
+            info['filename'] = info['file_name']
+            data_infos.append(info)
+        return data_infos
+
+    def get_ann_info(self, idx):
+        """Get COCO annotation by index.
+
+        Args:
+            idx (int): Index of data.
+
+        Returns:
+            dict: Annotation info of specified index.
+        """
+        img_id = self.data_infos[idx]['id']
+        ann_ids = self.coco.get_ann_ids(img_ids=[img_id])
+        ann_info = self.coco.load_anns(ann_ids)
+        # filter out unmatched images
+        ann_info = [i for i in ann_info if i['image_id'] == img_id]
+        return self._parse_ann_info(img_id, ann_info)
+
+    def _rgb_to_pan_id(self, png, bgr=True):
+        """Convert rgb to the annotation id.
+
+        Args:
+            png (numpy.ndarray): the ground truth map
+            bgr: whether the format of the color channel is 'bgr'.
+
+        Returns:
+            numpy.ndarray: the id map matched with the annotation file.
+        """
+        if bgr:
+            png = png[:, :, ::-1]
+        pan_id = png.astype(np.int32)
+        pan_id *= np.array([1, 256, 256 * 256]).reshape([1, 1, 3])
+        pan_id = np.sum(pan_id, axis=2)
+        return pan_id
+
+    def _parse_ann_info(self, image_id, ann_info):
+        """Parse annotations and load panoptic gts.
+
+        Args:
+            ann_info (list[dict]): Annotation info of an image.
+            image_id (int): The index of an image.
+
+        Returns:
+            dict: A dict containing the following keys: bboxes, bboxes_ignore,
+                labels, masks, seg_map.
+        """
+        gt_bboxes = []
+        gt_labels = []
+        gt_bboxes_ignore = []
+        gt_masks = []
+
+        pan_path = os.path.join(self.seg_prefix,
+                                self.formator.format(image_id))
+        assert os.path.isfile(pan_path), f'cannot find {pan_path}'
+        # todo: whether to use fileClient.
```

Comment:
We should do that. Therefore, we may simplify the logic in `_parse_ann_info` but add a new pipeline named `LoadPanopticAnnotations` that load the panoptic png and create gt bboxes, gt_masks and gt_semantic_seg.